src/core/models/column_caster.h

```diff
@@ -0,0 +1,77 @@
+//------------------------------------------------------------------------------
+// Copyright 2021 H2O.ai
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+//------------------------------------------------------------------------------
+
+#ifndef dt_MODELS_COLUMN_CASTER_h
+#define dt_MODELS_COLUMN_CASTER_h
+#include "column/func_unary.h"
+
+
+/**
+ *  Create a virtual column that casts numeric `col` from `T_from` to `T_to`.
+ *  Infinite values are casted into NA's.
+ */
+template <typename T_from, typename T_to>
+Column make_casted_column(const Column& col, dt::SType stype) {
+  Column col_casted = Column(new dt::FuncUnary2_ColumnImpl<T_from, T_to>(
+           Column(col),
+           [](T_from x, bool x_isvalid, T_to* out) {
+             *out = static_cast<T_to>(x);
+             return x_isvalid && _isfinite(x);
+           },
+           col.nrows(),
+           stype
+         ));
+
+  return col_casted;
+}
```

Comment:
How could I use it to convert infinity to NA?

Reply:
The reason I'm strongly suggesting using the standard casting functionality is because it's already relatively hard to implement efficiently, so having it in multiple places would just make it even harder.

Reply:
I guess I recall why I needed this inf check casting. The reason is that sometimes I need min/max, at least in the aggregator, and if there is an inf value it just spoils all the stats. Could be fixed if we had `min2` and `max2` functions implemented.